terraform/lambda-role.tf

```diff
@@ -30,3 +30,45 @@ resource "aws_iam_role_policy" "lambda_inline_policy" {
     }]
   })
 }
+
+
+resource "aws_iam_role" "control_plane_lambda_exec_role" {
+  name = "lambda-cognito-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Action = "sts:AssumeRole",
+        Effect = "Allow",
+        Principal = {
+          Service = "lambda.amazonaws.com"
+        }
+      }
+    ]
+  })
+}
+
+resource "aws_iam_policy" "control_plane_lambda_policy" {
+  name        = "lambda-user-management-policy"
+  description = "Policy for Lambda to manage usergroups"
+  policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [
+      {
+        Effect = "Allow",
+        Action = [
+          "cognito-idp:CreateGroup",
+          "cognito-idp:AdminAddUserToGroup",
+          "cognito-idp:InitiateAuth"
+        ],
+        Resource = "*"
+      }
+    ]
+  })
+}
+
+resource "aws_iam_role_policy_attachment" "attach_control_plane_lambda_policy" {
```

Comment:
Compare "resource "aws_iam_role_policy_attachment" "lambda_managed_policy""

This needs to be used to attach the AWS managed policy "arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole"

The custom inline policy for Cognito is already attached directly in control_plane_lambda_policy